packages/dev/core/src/Materials/PBR/openPbrMaterial.ts

```diff
@@ -0,0 +1,2550 @@
+        return true;
+    }
+
+    /**
+     * Force normal to face away from face.
+     * @internal
+     */
+    public _forceNormalForward = false;
+
+    /**
+     * Enables specular anti aliasing in the PBR shader.
+     * It will both interacts on the Geometry for analytical and IBL lighting.
+     * It also prefilter the roughness map based on the normalmap values.
+     * @internal
+     */
+    public _enableSpecularAntiAliasing = false;
+
+    /**
+     * Stores the available render targets.
+     */
+    private _renderTargets = new SmartArray<RenderTargetTexture>(16);
+
+    /**
+     * If set to true, no lighting calculations will be applied.
+     */
+    private _unlit = false;
+
+    /**
+     * If sets to true, the decal map will be applied after the detail map. Else, it is applied before (default: false)
+     */
+    private _applyDecalMapAfterDetailMap = false;
+
+    private _debugMode = 0;
+
+    private _shadersLoaded = false;
+    private _breakShaderLoadedCheck = false;
+
+    /**
+     * @internal
+     * This is reserved for the inspector.
+     * Defines the material debug mode.
+     * It helps seeing only some components of the material while troubleshooting.
+     */
+    @expandToProperty("_markAllSubMeshesAsMiscDirty")
+    public debugMode = 0;
+
+    /**
+     * @internal
+     * This is reserved for the inspector.
+     * Specify from where on screen the debug mode should start.
+     * The value goes from -1 (full screen) to 1 (not visible)
+     * It helps with side by side comparison against the final render
+     * This defaults to -1
+     */
+    public debugLimit = -1;
+
+    /**
+     * @internal
+     * This is reserved for the inspector.
+     * As the default viewing range might not be enough (if the ambient is really small for instance)
+     * You can use the factor to better multiply the final value.
+     */
+    public debugFactor = 1;
+
+    /**
+     * Defines additional PrePass parameters for the material.
+     */
+    public readonly prePassConfiguration: PrePassConfiguration;
+
+    protected _cacheHasRenderTargetTextures = false;
+
+    /**
+     * Instantiates a new OpenPBRMaterial instance.
+     *
+     * @param name The material name
+     * @param scene The scene the material will be use in.
+     * @param forceGLSL Use the GLSL code generation for the shader (even on WebGPU). Default is false
+     */
+    constructor(name: string, scene?: Scene, forceGLSL = false) {
+        super(name, scene, undefined, forceGLSL || PBRBaseMaterial.ForceGLSL);
+        // TODO: Check if we're running WebGL 2.0 or above
+        if (scene && !scene?.getEngine().isWebGPU && (scene.getEngine() as Engine).webGLVersion < 2) {
+            Logger.Error("OpenPBRMaterial: WebGL 2.0 or above is required for this material.");
+        }
+
+        if (OpenPBRMaterial._noiseTexture === null || OpenPBRMaterial._noiseTexture.getScene()?.getEngine() !== scene?.getEngine()) {

```

Comment:
`OpenPBRMaterial._noiseTexture` should probably be something like `{ [id: number]: BaseTexture }`, to avoid overwriting the old value with a new value if the scene is not the same than previously (you can access the map with `scene.uniqueId`).

Reply:
Done